### PR TITLE
Tarodane/scene port

### PR DESCRIPTION
**Pull request by Tarodane**
Added Scene+it's classes and Scene2dInstance

### CODE REVIEW

File: Scene/Scene2d.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+﻿using RenderSharp.RendererCommon;
+using RenderSharp.Math;
+
+namespace RenderSharp.Scene
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:5]: NIT: Fix extraneous spaces

File: Scene/Scene2d.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+{
+    public class Scene2d
+    {
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:9]: here

File: Scene/Scene2d.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        public class Camera
+        {
+            public Vec2 Center { get; set; }
+            public double Zoom { get; set; }
+            public double Rotation { get; set; }
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:15]: here

File: Scene/Scene2d.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            {
+                Zoom += zoom;
+            }
+
+            public void Rotate(double radChange) 
+            {
+                Rotation = radChange;
+            }
+        }
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:38]: here

File: Scene/Scene2d.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            {
+                Size = (Vec2) ((FVec2) Size * scale);
+            }
+
+            public void ClearShaders()
+            {
+                Shader = (FragShaderArgs) => { }; 
+            }
+        }
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:75]: here

File: Scene/Scene2dInstance.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+﻿using RenderSharp.Math;
+using RenderSharp.RendererCommon;
+
+namespace RenderSharp.Scene
+{
+    internal class Scene2dInstance
```
Comment (LilTOJustice) [inline Scene/Scene2dInstance.cs:6]: This should be public so other classes can access

File: Scene/Scene2d.cs [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            BgColor = bgcolor ?? new RGB();
+            BgTexture = texture ?? new Texture(new Vec2(1, 1));
+            Duration = duration;
+
+            TimeSeq = new List<double>();
+            for (int i = 0; i < framerate*duration; i++)
+            {
+                TimeSeq.Add(i*DeltaTime);
+            }
+
```
Comment (LilTOJustice) [inline Scene/Scene2d.cs:99]: here

**Review by LilTOJustice — CHANGES_REQUESTED** [review]
Couple small suggestions first

File: Scene/Scene2dInstance.cs (hunk 6 above)
Comment (Tarodane) [reply Scene/Scene2dInstance.cs:6]: Do I remove "internal" and replace it with "public"?

**Review by LilTOJustice — CHANGES_REQUESTED** [review]
SceneThinkFunc is missing!

**Review by LilTOJustice — APPROVED** [review]

**Pull request merged**